bbl_screen-patch/kexec_ui/printerui/qml/dialog/KexecDialog.qml

```diff
@@ -16,34 +17,78 @@ Item {
         return xhr.status == 200;
     }())
     property var countdown: 10
+    property string currentMode: (x1pName !== "") ? "ota" : "default"
+     
+    property var startupStrings: {
+        "default": { /* Normal Boot: 10 sec dialog then boot X1Plus */
+            "buttons": {
+                "yes_confirm": {"text": qsTr("Boot X1Plus"), "action": function() { X1PlusNative.system("/opt/kexec/boot"); }},
+                "no": {"text": qsTr("Startup options..."), "action": function() { dialogStack.replace("../BootOptionsPage.qml", {hasSdCard: hasSdCard}) }},
+                "cancel": {"text": "", "action": function() {}},  
+            },
+            "title": qsTr("Bootable SD card detected."),
+            "subtitle": function() {
+                if (countdown > 0)
+                    return qsTr("Your printer will automatically boot from the SD card in %1 seconds.").arg(countdown)
+                else 
+                    return qsTr("Your printer is rebooting into the OS on the inserted SD card.")
+                    
+            }
+        },
+        "ota": {
+            "buttons": {/* auto install x1p Boot: 10 sec dialog then tell InstallingPage.qml to start installing x1p */
+                "yes_confirm": {"text": qsTr("Install"), "action": function() { dialogStack.replace("../InstallingPage.qml", {x1pName: x1pName}) }},
+                "no": {"text": qsTr("Startup options"), "action": function() { dialogStack.replace("../BootOptionsPage.qml", {hasSdCard: hasSdCard}) }},
+                "cancel": {"text": qsTr("Boot X1Plus"), "action": function() { X1PlusNative.system("/opt/kexec/boot"); }}
+            },
+            "title": qsTr("Update ready to install!"),
+            "subtitle": function() {
+                if (countdown > 0)
+                    return qsTr("Automatically installing .x1p file '%1' in %2 seconds. To cancel or skip installation, select 'Startup options' or 'Boot X1Plus'.").arg(x1pName).arg(countdown);
```

Comment:
I will tweak the wording of this.  also I wonder if we should just have 'install | startup options'